src/main/scala/beam/agentsim/infrastructure/ChargingNetwork.scala

```diff
@@ -129,13 +130,16 @@ object ChargingNetwork {
       mutable.PriorityQueue.empty[ChargingVehicle](Ordering.by((_: ChargingVehicle).arrivalTime).reverse)
 
     def numAvailableChargers: Int = zone.numChargers - connectedVehiclesInternal.size
-    def connectedVehicles: Map[Id[BeamVehicle], ChargingVehicle] = connectedVehiclesInternal.toMap
+    def connectedVehicles: collection.Map[Id[BeamVehicle], ChargingVehicle] = connectedVehiclesInternal
```

Comment:
The reason (as far as I can tell) this was passed as `toMap` was to make sure there would be no mutation of the underlying backing. I would not suggest this change..

Reply:
You are right, though it's used only for reading. Though to avoid mutation in the future I restored that toMap and minimized the number of calls to this method.